cpu/msp430_common/include/irq_arch.h

```diff
@@ -83,12 +84,12 @@ __attribute__((always_inline)) static inline void irq_restore(unsigned int state
     );
 }
 
-__attribute__((always_inline)) static inline int irq_is_in(void)
+__attribute__((always_inline)) static inline bool irq_is_in(void)
 {
     return __irq_is_in;
 }
 
-__attribute__((always_inline)) static inline int irq_is_enabled(void)
+__attribute__((always_inline)) static inline bool irq_is_enabled(void)
 {
     unsigned int state;
     __asm__ volatile(
```

Comment:
`vendor/msp430-gcc-support-files/include/msp430f1611.h:#define GIE                    (0x0008)`

so, `state & GIE` is either 0 or 0x8 ...

Reply:
so?

Reply:
The compiler will convert any non-zero value to `1` if the type is `bool`. Note that with static inline functions this conversion can be optimised out if it is not needed (e.g. a branch if non-zero is used on the return value).

I tested a few years ago of using `bool` over `int` adds overhead for `static inline` functions, and for my test codes this wasn't the case.